### ISSUE TITLE
Gemma example does not complete successfully for GPU or CUDA. 

### ISSUE THREAD

**Issue by danielclough** [huggingface/candle#1752]
On Ubuntu 22.04 running CUDA and CPU it hangs in the same place.

Repo is up to date with `main`.

Using default example prompt:
```sh
cargo run --example gemma --release -- --prompt "fn count_primes(max_n: usize)"
```

```rs
loaded the model in 2.828576967s
fn count_primes(max_n: usize) -> usize {
    let mut primes = vec![true; max_n];
    for i in 2..=max_n {
        if primes[i] {
            for j in i * i..max_n {
                primes[j] = false;
             }
         }
    }
    primes.len()
}

fn main() {
    println!("{}", count_primes
```

**Comment by LaurentMazare** [comment]
Should be fixed in #1753 , thanks for reporting this.

**Comment by danielclough** [comment]
You are my Hero! :supervillain:
